includes/data-types/class-payment-information.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Class Payment_Information
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\DataTypes;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Mostly a wrapper containing information on a single payment.
+ */
+class Payment_Information {
+	/**
+	 * The ID of the payment method used for this payment.
+	 *
+	 * @var string
+	 */
+	private $payment_method;
+
+	/**
+	 * The payment token used for this payment.
+	 *
+	 * @var \WC_Payment_Token/NULL
+	 */
+	private $token;
+
+	/**
+	 * Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
+	 *
+	 * @var bool
+	 */
+	private $off_session;
+
+	/**
+	 * Payment information constructor.
+	 *
+	 * @param string            $payment_method The ID of the payment method used for this payment.
+	 * @param \WC_Payment_Token $token The payment token used for this payment.
+	 * @param bool              $off_session Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
+	 *
+	 * @throws \Exception - If no payment method is found in the provided request.
+	 */
+	public function __construct(
+		string $payment_method,
+		\WC_Payment_Token $token = null,
+		bool $off_session = false
+	) {
```

Comment:
I want to get rid of the `bool` flag in the constructor since `new Payment_Information( $payment_method, $payment_token, true )` makes a lot less sense than `new Payment_Information( $payment_method, $payment_token, MERCHANT_INITIATED )`.

I'll get on this tomorrow.

Reply:
Moved to #860